server/model/tests_test.go

```diff
@@ -0,0 +1,157 @@
+package model_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/kubeshop/tracetest/server/assertions/comparator"
+	"github.com/kubeshop/tracetest/server/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDefinition(t *testing.T) {
+	t.Run("Add", func(t *testing.T) {
+		def := (model.Test{}).Definition
+
+		def, err := def.Add(model.SpanQuery("1"), []model.Assertion{{"1", comparator.Eq, "1"}})
+		require.NoError(t, err)
+
+		def, err = def.Add(model.SpanQuery("2"), []model.Assertion{{"2", comparator.Eq, "2"}})
+		require.NoError(t, err)
+		assert.Equal(t, 2, def.Len())
+
+		def, err = def.Add(model.SpanQuery("2"), []model.Assertion{{"2", comparator.Eq, "2"}})
+		assert.ErrorContains(t, err, "selector already exists")
+		assert.Equal(t, 0, def.Len())
+
+	})
+
+	generateDef := func() model.OrderedMap[model.SpanQuery, []model.Assertion] {
+		def := (model.Test{}).Definition
+
+		def, _ = def.Add(model.SpanQuery("1"), []model.Assertion{{"1", comparator.Eq, "1"}})
+		def, _ = def.Add(model.SpanQuery("2"), []model.Assertion{{"2", comparator.Eq, "2"}})
+
+		return def
+	}
```

Comment:
You are duplicating that function. Instead of creating an anonymous function, create it outside the test functions so you can reuse the same function in both tests.

Reply:
It's not exactly the same. This function is being used in the `Map` and `Get` tests. The `Add` test is not exactly the same, since it also checks the errors.

Reply:
You're right.